<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1073</width>
    <height>574</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="load">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>350</y>
      <width>101</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>load image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="close">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>430</y>
      <width>101</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>close</string>
    </property>
   </widget>
   <widget class="QLabel" name="showImageLabel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>300</width>
      <height>300</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:rgb(170, 170, 255)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="salt">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>340</y>
      <width>170</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>add Salt Pepper noise</string>
    </property>
   </widget>
   <widget class="QPushButton" name="gaussian">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>340</y>
      <width>170</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>add Gaussian noise</string>
    </property>
   </widget>
   <widget class="QLabel" name="PSNR">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>400</y>
      <width>57</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>PSNR:</string>
    </property>
   </widget>
   <widget class="QLabel" name="MMSIM">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>420</y>
      <width>57</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>MSSIM:</string>
    </property>
   </widget>
   <widget class="QLabel" name="P_num">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>400</y>
      <width>57</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="b">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>490</y>
      <width>57</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="g">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>465</y>
      <width>57</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="r">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>440</y>
      <width>57</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="showImageLabel_2">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>30</y>
      <width>300</width>
      <height>300</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:rgb(170, 170, 255)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="blur">
    <property name="geometry">
     <rect>
      <x>611</x>
      <y>340</y>
      <width>141</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Average Blur</string>
    </property>
   </widget>
   <widget class="QPushButton" name="g_blur">
    <property name="geometry">
     <rect>
      <x>611</x>
      <y>380</y>
      <width>141</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Gaussian Blur</string>
    </property>
   </widget>
   <widget class="QPushButton" name="m_blur">
    <property name="geometry">
     <rect>
      <x>611</x>
      <y>420</y>
      <width>141</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Median Blur</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>440</y>
      <width>57</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>r:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>465</y>
      <width>57</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>g:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>490</y>
      <width>57</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>b:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="salt_combo">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>370</y>
      <width>170</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>100</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>1000</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>10000</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="para">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>340</y>
      <width>141</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>calculate(before blur)</string>
    </property>
   </widget>
   <widget class="QComboBox" name="mu">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>370</y>
      <width>82</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>0</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>10</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>20</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>30</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>40</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>100</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="sigma">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>370</y>
      <width>82</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>0</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>10</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>20</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>30</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>40</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>100</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="showImageLabel_3">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>30</y>
      <width>300</width>
      <height>300</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:rgb(170, 170, 255)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>0</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>source</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>0</y>
      <width>90</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>add noise</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>0</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>blur result</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="para_2">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>370</y>
      <width>141</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>calculate(after blur)</string>
    </property>
   </widget>
   <widget class="QComboBox" name="b_combo">
    <property name="geometry">
     <rect>
      <x>761</x>
      <y>340</y>
      <width>82</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>3*3</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>5*5</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>8*8</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="g_combo">
    <property name="geometry">
     <rect>
      <x>761</x>
      <y>380</y>
      <width>82</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>3*3</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>5*5</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>7*7</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="m_combo">
    <property name="geometry">
     <rect>
      <x>761</x>
      <y>420</y>
      <width>82</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>3</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>5</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>7</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="exponential">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>420</y>
      <width>170</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>add Exponential noise</string>
    </property>
   </widget>
   <widget class="QComboBox" name="ex_combo">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>450</y>
      <width>170</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>1</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>10</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>100</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="random">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>420</y>
      <width>170</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>add Random noise</string>
    </property>
   </widget>
   <widget class="QPushButton" name="export_file">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>450</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>export</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bilater">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>460</y>
      <width>141</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Bilateral Filter</string>
    </property>
   </widget>
   <widget class="QPushButton" name="c_av">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>340</y>
      <width>80</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Average_</string>
    </property>
   </widget>
   <widget class="QPushButton" name="c_med">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>420</y>
      <width>80</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Median_</string>
    </property>
   </widget>
   <widget class="QPushButton" name="c_gau">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>380</y>
      <width>80</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Gaussian_</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1073</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
